client/my-sites/site-settings/form-seo.jsx

```diff
@@ -84,19 +84,26 @@ export const SeoForm = React.createClass( {
 		return stateForSite( this.props.site );
 	},
 
-	componentWillReceiveProps: function( nextProps ) {
+	componentDidMount() {
+		this.redirectJetpack( this.props.site );
```

Comment:
We tend to do redirects like this at the controller level (`my-sites/site-settings/controller.js`). There's even a `redirectToGeneral` function there for this purpose.